### PR TITLE
Add Tableau connector source files

### PR DESCRIPTION
**Pull request by rupal-bq**
*Issue #81:*

*Description of changes:*
- Added connection dialog
- Added dialect & customizations 

Capture

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/TaCo/connection-dialog.tcd [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<connection-dialog class='odfe_sql_odbc'>
+  <connection-config>  
+    <authentication-mode value='ComboBoxIntegrated' />
+        <authentication-options>
+            <option name="None" value="auth-none" default="true" />
```
Comment (jordanw-bq) [inline src/TaCo/connection-dialog.tcd:5]: Is it possible to set this value to `NONE` so we can directly parse the value as our connection string `Auth` parameter?
```suggestion
            <option name="None" value="NONE" default="true" />
```

File: src/TaCo/connectionBuilder.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+(function dsbuilder(attr){
+    var params = {};
+
+    // Set host information in connection string
+    params["SERVER"] = attr[connectionHelper.attributeServer];
+    params["PORT"] = attr[connectionHelper.attributePort];
+
+    // Set authentication values in connection string
+    var authAttrValue = attr[connectionHelper.attributeAuthentication];
+    if (authAttrValue == "auth-none"){
```
Comment (jordanw-bq) [inline src/TaCo/connectionBuilder.js:10]: If we can set the auth option in `connection-dialog.tcd`, we can simplify this logic here

File: src/TaCo/connectionBuilder.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    } else if (authAttrValue == "BASIC"){
+        params["Auth"] = "BASIC";
+        params["UID"] = attr[connectionHelper.attributeUsername];
+        params["PWD"] = attr[connectionHelper.attributePassword];
+    }
+
+    // Set SSL value in connection string 
+    if (attr[connectionHelper.attributeSSLMode] == "require"){
+        params["useSSL"] = "1";
+    }
```
Comment (jordanw-bq) [inline src/TaCo/connectionBuilder.js:24]: Let's explicitly set this to `"0"` in an `else` case

**Comment by jordanw-bq** [comment]
Also, can we change `src/TaCo/` to `src/TableauConnector`? I do like the abbreviation, but I think we should stay on the side of clarity here.

File: src/TaCo/manifest.xml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      <support-link url="https://github.com/opendistro-for-elasticsearch/sql-odbc"/>
+      <driver-download-link url="https://github.com/opendistro-for-elasticsearch/sql-odbc"/>
+  </vendor-information>
+  <connection-customization class="odfe_sql_odbc" enabled="true" version="10.0">
+    <vendor name="Amazon"/>
+    <driver name="ODFE SQL ODBC Driver"/>
+    <customizations>
+      <customization name="CAP_CREATE_TEMP_TABLES" value="no"/>
+      <customization name="CAP_SUPPRESS_DISCOVERY_QUERIES" value="yes" />
+      <customization name="SQL_IDENTIFIER_QUOTE_CHAR" value="`" />
```
[the remark this answers is not in the document]
Comment (rupal-bq) [reply src/TaCo/manifest.xml]: Removed unused customization

File: src/TaCo/connection-dialog.tcd (hunk 1 above)
Comment (rupal-bq) [reply src/TaCo/connection-dialog.tcd]: done

File: src/TaCo/connectionBuilder.js (hunk 2 above)
Comment (rupal-bq) [reply src/TaCo/connectionBuilder.js]: done

File: src/TaCo/connectionBuilder.js (hunk 3 above)
Comment (rupal-bq) [reply src/TaCo/connectionBuilder.js]: done

**Comment by rupal-bq** [comment]
> Also, can we change `src/TaCo/` to `src/TableauConnector`? I do like the abbreviation, but I think we should stay on the side of clarity here.

Renamed to `src/TableauConnector`

**Pull request merged**